### PR TITLE
F aws add multiple listeners and port matching

### LINKED ISSUE #26380: aws_appmesh_virtual_node should support multiple listeners
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

Via the console it is possible to specify multiple listeners for an app mesh virtual node.

The terraform resource `aws_appmesh_virtual_node` seems to only support the ability to define a single listener. Even when importing a node with multiple listeners defined, it only creates configuration for the first listener in the list.

### New or Affected Resource(s)

* aws_appmesh_virtual_node

### Potential Terraform Configuration

```hcl
resource "aws_appmesh_virtual_node" "serviceb1" {
  name      = "serviceBv1"
  mesh_name = aws_appmesh_mesh.simple.id

  spec {
    backend {
      virtual_service {
        virtual_service_name = "servicea.simpleapp.local"
      }
    }

    listener {
      port_mapping {
        port     = 8081
        protocol = "http"
      }
    }

    listener {
      port_mapping {
        port     = 8080
        protocol = "http"
      }
    }

    service_discovery {
      dns {
        hostname = "serviceb.simpleapp.local"
      }
    }
  }
}
```

### LINKED ISSUE #27698: [Bug/Enhancement]: appmesh_route needs support for port matching
### Description

app mesh routes seem to require setting port matching on the route. I actually can't get it to work at all without doing this, which means that this resource is currently useless?

### Affected Resource(s) and/or Data Source(s)

appmesh_route 

### Potential Terraform Configuration

```terraform
resource "aws_appmesh_route" "routes" {
  name                = var.service_name
  mesh_name           = var.mesh_name
  virtual_router_name = var.mesh_virtual_router_name

  spec {
    http_route {
      match {
        prefix = "/"
        router_port = 8080 # new addition
      }

      action {
        weighted_target {
          virtual_node = aws_appmesh_virtual_node.node[0].name
          weight       = 100
          target_port  = 80 # new addition
        }
      }
    }
  }
}
```

### References

error when adding router listeners:

image

port config from gui:
image

### Would you like to implement a fix?

_No response_

### PR DESCRIPTION
**Pull request by dgr237**
### Description
This Pull Request adds the functionality to be able specify multi listeners on the following resources:

aws_appmesh_virtual_node
aws_appmesh_virtual_gateway
aws_appmesh_virtual_router

In addition port has been added to match on the routes for:

aws_appmesh_gateway_route
aws_appmesh_route

Added weighted_target port to aws_appmesh_route.

### Relations
Closes #27698
Closes #26380

### References
This functionality enables services to expose multiple ports/ protocols and route traffic to the different ports form other services running on the cluster or from the gateway.

### Output from Acceptance Testing

```
$ make testacc PKG=appmesh          
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./internal/service/appmesh/... -v -count 1 -parallel 20   -timeout 180m
=== RUN   TestAccAppMesh_serial
=== RUN   TestAccAppMesh_serial/Route
=== RUN   TestAccAppMesh_serial/Route/httpRoute
=== RUN   TestAccAppMesh_serial/Route/tcpRouteWithPortMatch
=== RUN   TestAccAppMesh_serial/Route/grpcRouteEmptyMatch
=== RUN   TestAccAppMesh_serial/Route/http2RouteWithPortMatch
=== RUN   TestAccAppMesh_serial/Route/tcpRoute
=== RUN   TestAccAppMesh_serial/Route/grpcRouteWithPortMatch
=== RUN   TestAccAppMesh_serial/Route/http2Route
=== RUN   TestAccAppMesh_serial/Route/httpRouteWithPortMatch
=== RUN   TestAccAppMesh_serial/Route/httpRouteTimeout
=== RUN   TestAccAppMesh_serial/Route/routePriority
=== RUN   TestAccAppMesh_serial/Route/tags
=== RUN   TestAccAppMesh_serial/Route/grpcRoute
=== RUN   TestAccAppMesh_serial/Route/httpRetryPolicy
=== RUN   TestAccAppMesh_serial/Route/httpHeader
=== RUN   TestAccAppMesh_serial/Route/tcpRouteTimeout
=== RUN   TestAccAppMesh_serial/Route/grpcRouteTimeout
=== RUN   TestAccAppMesh_serial/Route/http2RouteTimeout
=== RUN   TestAccAppMesh_serial/VirtualGateway
=== RUN   TestAccAppMesh_serial/VirtualGateway/disappears
=== RUN   TestAccAppMesh_serial/VirtualGateway/listenerHealthChecks
=== RUN   TestAccAppMesh_serial/VirtualGateway/logging
=== RUN   TestAccAppMesh_serial/VirtualGateway/listenerTls
=== RUN   TestAccAppMesh_serial/VirtualGateway/listenerValidation
=== RUN   TestAccAppMesh_serial/VirtualGateway/multiListenerValidation
=== RUN   TestAccAppMesh_serial/VirtualGateway/tags
=== RUN   TestAccAppMesh_serial/VirtualGateway/basic
=== RUN   TestAccAppMesh_serial/VirtualGateway/backendDefaults
=== RUN   TestAccAppMesh_serial/VirtualGateway/backendDefaultsCertificate
=== RUN   TestAccAppMesh_serial/VirtualGateway/listenerConnectionPool
=== RUN   TestAccAppMesh_serial/VirtualNode
=== RUN   TestAccAppMesh_serial/VirtualNode/backendDefaultsCertificate
=== RUN   TestAccAppMesh_serial/VirtualNode/listenerOutlierDetection
=== RUN   TestAccAppMesh_serial/VirtualNode/listenerValidation
=== RUN   TestAccAppMesh_serial/VirtualNode/logging
=== RUN   TestAccAppMesh_serial/VirtualNode/basic
=== RUN   TestAccAppMesh_serial/VirtualNode/backendDefaults
=== RUN   TestAccAppMesh_serial/VirtualNode/cloudMapServiceDiscovery
=== RUN   TestAccAppMesh_serial/VirtualNode/tags
=== RUN   TestAccAppMesh_serial/VirtualNode/multiListenerValidation
=== RUN   TestAccAppMesh_serial/VirtualNode/disappears
=== RUN   TestAccAppMesh_serial/VirtualNode/backendClientPolicyAcm
=== RUN   TestAccAppMesh_serial/VirtualNode/backendClientPolicyFile
=== RUN   TestAccAppMesh_serial/VirtualNode/listenerConnectionPool
=== RUN   TestAccAppMesh_serial/VirtualNode/listenerTimeout
=== RUN   TestAccAppMesh_serial/VirtualNode/listenerHealthChecks
=== RUN   TestAccAppMesh_serial/VirtualNode/listenerTls
=== RUN   TestAccAppMesh_serial/VirtualRouter
=== RUN   TestAccAppMesh_serial/VirtualRouter/basic
=== RUN   TestAccAppMesh_serial/VirtualRouter/multiListener
=== RUN   TestAccAppMesh_serial/VirtualRouter/tags
=== RUN   TestAccAppMesh_serial/VirtualService
=== RUN   TestAccAppMesh_serial/VirtualService/virtualNode
=== RUN   TestAccAppMesh_serial/VirtualService/virtualRouter
=== RUN   TestAccAppMesh_serial/VirtualService/tags
=== RUN   TestAccAppMesh_serial/GatewayRoute
=== RUN   TestAccAppMesh_serial/GatewayRoute/grpcRoute
=== RUN   TestAccAppMesh_serial/GatewayRoute/grpcRouteWithPort
=== RUN   TestAccAppMesh_serial/GatewayRoute/httpRoute
=== RUN   TestAccAppMesh_serial/GatewayRoute/tags
=== RUN   TestAccAppMesh_serial/GatewayRoute/basic
=== RUN   TestAccAppMesh_serial/GatewayRoute/disappears
=== RUN   TestAccAppMesh_serial/GatewayRoute/httpRouteWithPort
=== RUN   TestAccAppMesh_serial/GatewayRoute/http2Route
=== RUN   TestAccAppMesh_serial/GatewayRoute/http2RouteWithPort
=== RUN   TestAccAppMesh_serial/Mesh
=== RUN   TestAccAppMesh_serial/Mesh/basic
=== RUN   TestAccAppMesh_serial/Mesh/egressFilter
=== RUN   TestAccAppMesh_serial/Mesh/tags
--- PASS: TestAccAppMesh_serial (2355.27s)
    --- PASS: TestAccAppMesh_serial/Route (656.74s)
        --- PASS: TestAccAppMesh_serial/Route/httpRoute (46.00s)
        --- PASS: TestAccAppMesh_serial/Route/tcpRouteWithPortMatch (44.08s)
        --- PASS: TestAccAppMesh_serial/Route/grpcRouteEmptyMatch (19.56s)
        --- PASS: TestAccAppMesh_serial/Route/http2RouteWithPortMatch (31.98s)
        --- PASS: TestAccAppMesh_serial/Route/tcpRoute (43.91s)
        --- PASS: TestAccAppMesh_serial/Route/grpcRouteWithPortMatch (56.64s)
        --- PASS: TestAccAppMesh_serial/Route/http2Route (31.86s)
        --- PASS: TestAccAppMesh_serial/Route/httpRouteWithPortMatch (44.22s)
        --- PASS: TestAccAppMesh_serial/Route/httpRouteTimeout (32.07s)
        --- PASS: TestAccAppMesh_serial/Route/routePriority (31.77s)
        --- PASS: TestAccAppMesh_serial/Route/tags (44.52s)
        --- PASS: TestAccAppMesh_serial/Route/grpcRoute (56.25s)
        --- PASS: TestAccAppMesh_serial/Route/httpRetryPolicy (44.28s)
        --- PASS: TestAccAppMesh_serial/Route/httpHeader (31.55s)
        --- PASS: TestAccAppMesh_serial/Route/tcpRouteTimeout (31.58s)
        --- PASS: TestAccAppMesh_serial/Route/grpcRouteTimeout (31.58s)
        --- PASS: TestAccAppMesh_serial/Route/http2RouteTimeout (34.89s)
    --- PASS: TestAccAppMesh_serial/VirtualGateway (364.04s)
        --- PASS: TestAccAppMesh_serial/VirtualGateway/disappears (15.92s)
        --- PASS: TestAccAppMesh_serial/VirtualGateway/listenerHealthChecks (32.84s)
        --- PASS: TestAccAppMesh_serial/VirtualGateway/logging (31.47s)
        --- PASS: TestAccAppMesh_serial/VirtualGateway/listenerTls (87.32s)
        --- PASS: TestAccAppMesh_serial/VirtualGateway/listenerValidation (29.62s)
        --- PASS: TestAccAppMesh_serial/VirtualGateway/multiListenerValidation (29.72s)
        --- PASS: TestAccAppMesh_serial/VirtualGateway/tags (40.72s)
        --- PASS: TestAccAppMesh_serial/VirtualGateway/basic (18.44s)
        --- PASS: TestAccAppMesh_serial/VirtualGateway/backendDefaults (29.84s)
        --- PASS: TestAccAppMesh_serial/VirtualGateway/backendDefaultsCertificate (18.36s)
        --- PASS: TestAccAppMesh_serial/VirtualGateway/listenerConnectionPool (29.79s)
    --- PASS: TestAccAppMesh_serial/VirtualNode (664.79s)
        --- PASS: TestAccAppMesh_serial/VirtualNode/backendDefaultsCertificate (18.56s)
        --- PASS: TestAccAppMesh_serial/VirtualNode/listenerOutlierDetection (30.49s)
        --- PASS: TestAccAppMesh_serial/VirtualNode/listenerValidation (30.69s)
        --- PASS: TestAccAppMesh_serial/VirtualNode/logging (30.12s)
        --- PASS: TestAccAppMesh_serial/VirtualNode/basic (19.29s)
        --- PASS: TestAccAppMesh_serial/VirtualNode/backendDefaults (31.37s)
        --- PASS: TestAccAppMesh_serial/VirtualNode/cloudMapServiceDiscovery (120.43s)
        --- PASS: TestAccAppMesh_serial/VirtualNode/tags (41.31s)
        --- PASS: TestAccAppMesh_serial/VirtualNode/multiListenerValidation (29.88s)
        --- PASS: TestAccAppMesh_serial/VirtualNode/disappears (15.94s)
        --- PASS: TestAccAppMesh_serial/VirtualNode/backendClientPolicyAcm (64.71s)
        --- PASS: TestAccAppMesh_serial/VirtualNode/backendClientPolicyFile (45.13s)
        --- PASS: TestAccAppMesh_serial/VirtualNode/listenerConnectionPool (40.15s)
        --- PASS: TestAccAppMesh_serial/VirtualNode/listenerTimeout (35.65s)
        --- PASS: TestAccAppMesh_serial/VirtualNode/listenerHealthChecks (29.95s)
        --- PASS: TestAccAppMesh_serial/VirtualNode/listenerTls (81.13s)
    --- PASS: TestAccAppMesh_serial/VirtualRouter (101.20s)
        --- PASS: TestAccAppMesh_serial/VirtualRouter/basic (30.00s)
        --- PASS: TestAccAppMesh_serial/VirtualRouter/multiListener (29.98s)
        --- PASS: TestAccAppMesh_serial/VirtualRouter/tags (41.21s)
    --- PASS: TestAccAppMesh_serial/VirtualService (104.38s)
        --- PASS: TestAccAppMesh_serial/VirtualService/virtualNode (31.60s)
        --- PASS: TestAccAppMesh_serial/VirtualService/virtualRouter (28.76s)
        --- PASS: TestAccAppMesh_serial/VirtualService/tags (44.02s)
    --- PASS: TestAccAppMesh_serial/GatewayRoute (372.20s)
        --- PASS: TestAccAppMesh_serial/GatewayRoute/grpcRoute (32.32s)
        --- PASS: TestAccAppMesh_serial/GatewayRoute/grpcRouteWithPort (31.98s)
        --- PASS: TestAccAppMesh_serial/GatewayRoute/httpRoute (56.33s)
        --- PASS: TestAccAppMesh_serial/GatewayRoute/tags (43.98s)
        --- PASS: TestAccAppMesh_serial/GatewayRoute/basic (19.86s)
        --- PASS: TestAccAppMesh_serial/GatewayRoute/disappears (17.07s)
        --- PASS: TestAccAppMesh_serial/GatewayRoute/httpRouteWithPort (56.34s)
        --- PASS: TestAccAppMesh_serial/GatewayRoute/http2Route (56.60s)
        --- PASS: TestAccAppMesh_serial/GatewayRoute/http2RouteWithPort (57.72s)
    --- PASS: TestAccAppMesh_serial/Mesh (91.92s)
        --- PASS: TestAccAppMesh_serial/Mesh/basic (17.86s)
        --- PASS: TestAccAppMesh_serial/Mesh/egressFilter (35.12s)
        --- PASS: TestAccAppMesh_serial/Mesh/tags (38.95s)
=== RUN   TestAccAppMeshMeshDataSource_basic
=== PAUSE TestAccAppMeshMeshDataSource_basic
=== RUN   TestAccAppMeshMeshDataSource_meshOwner
=== PAUSE TestAccAppMeshMeshDataSource_meshOwner
=== RUN   TestAccAppMeshMeshDataSource_specAndTagsSet
=== PAUSE TestAccAppMeshMeshDataSource_specAndTagsSet
=== RUN   TestVirtualNodeMigrateState
--- PASS: TestVirtualNodeMigrateState (0.00s)
=== RUN   TestVirtualRouterMigrateState
--- PASS: TestVirtualRouterMigrateState (0.00s)
=== RUN   TestAccAppMeshVirtualServiceDataSource_virtualNode
=== PAUSE TestAccAppMeshVirtualServiceDataSource_virtualNode
=== RUN   TestAccAppMeshVirtualServiceDataSource_virtualRouter
=== PAUSE TestAccAppMeshVirtualServiceDataSource_virtualRouter
=== CONT  TestAccAppMeshMeshDataSource_basic
=== CONT  TestAccAppMeshVirtualServiceDataSource_virtualNode
=== CONT  TestAccAppMeshMeshDataSource_specAndTagsSet
=== CONT  TestAccAppMeshMeshDataSource_meshOwner
=== CONT  TestAccAppMeshVirtualServiceDataSource_virtualRouter
--- PASS: TestAccAppMeshMeshDataSource_specAndTagsSet (18.27s)
--- PASS: TestAccAppMeshMeshDataSource_basic (19.15s)
--- PASS: TestAccAppMeshMeshDataSource_meshOwner (20.36s)
--- PASS: TestAccAppMeshVirtualServiceDataSource_virtualNode (21.27s)
--- PASS: TestAccAppMeshVirtualServiceDataSource_virtualRouter (21.36s)
PASS
ok      [github.com/hashicorp/terraform-provider-aws/internal/service/appmesh](http://github.com/hashicorp/terraform-provider-aws/internal/service/appmesh)    2378.501s

...
```

### CODE REVIEW

**Review by bendu — APPROVED** [review]

**Comment by bendu** [comment]
@YakDriver @breathingdust can you please take a look at this PR? I'm from the AWS App Mesh team and the change seems fine to me.

**Review by bendu — APPROVED** [review]

**Review by johnsonaj — APPROVED** [review]
LGTM 🚀 

```console
$ make testacc TESTARGS='-run=TestAccAppMesh_serial' PKG=appmesh

==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./internal/service/appmesh/... -v -count 1 -parallel 20  -run=TestAccAppMesh_serial -timeout 180m
--- PASS: TestAccAppMesh_serial (1744.11s)
    --- PASS: TestAccAppMesh_serial/VirtualRouter (81.56s)
        --- PASS: TestAccAppMesh_serial/VirtualRouter/tags (29.91s)
        --- PASS: TestAccAppMesh_serial/VirtualRouter/basic (20.59s)
        --- PASS: TestAccAppMesh_serial/VirtualRouter/disappears (10.45s)
        --- PASS: TestAccAppMesh_serial/VirtualRouter/multiListener (20.62s)
    --- PASS: TestAccAppMesh_serial/VirtualService (85.33s)
        --- PASS: TestAccAppMesh_serial/VirtualService/disappears (11.35s)
        --- PASS: TestAccAppMesh_serial/VirtualService/virtualNode (22.52s)
        --- PASS: TestAccAppMesh_serial/VirtualService/virtualRouter (20.13s)
        --- PASS: TestAccAppMesh_serial/VirtualService/tags (31.32s)
    --- PASS: TestAccAppMesh_serial/GatewayRoute (263.11s)
        --- PASS: TestAccAppMesh_serial/GatewayRoute/http2Route (40.41s)
        --- PASS: TestAccAppMesh_serial/GatewayRoute/http2RouteWithPort (40.48s)
        --- PASS: TestAccAppMesh_serial/GatewayRoute/basic (13.48s)
        --- PASS: TestAccAppMesh_serial/GatewayRoute/grpcRoute (22.44s)
        --- PASS: TestAccAppMesh_serial/GatewayRoute/httpRoute (40.53s)
        --- PASS: TestAccAppMesh_serial/GatewayRoute/httpRouteWithPort (40.52s)
        --- PASS: TestAccAppMesh_serial/GatewayRoute/tags (31.34s)
        --- PASS: TestAccAppMesh_serial/GatewayRoute/disappears (11.44s)
        --- PASS: TestAccAppMesh_serial/GatewayRoute/grpcRouteWithPort (22.46s)
    --- PASS: TestAccAppMesh_serial/Mesh (63.75s)
        --- PASS: TestAccAppMesh_serial/Mesh/basic (11.69s)
        --- PASS: TestAccAppMesh_serial/Mesh/egressFilter (24.50s)
        --- PASS: TestAccAppMesh_serial/Mesh/tags (27.56s)
    --- PASS: TestAccAppMesh_serial/Route (517.77s)
        --- PASS: TestAccAppMesh_serial/Route/http2RouteWithPortMatch (22.58s)
        --- PASS: TestAccAppMesh_serial/Route/httpRetryPolicy (31.42s)
        --- PASS: TestAccAppMesh_serial/Route/tcpRouteWithPortMatch (31.74s)
        --- PASS: TestAccAppMesh_serial/Route/tcpRouteTimeout (22.57s)
        --- PASS: TestAccAppMesh_serial/Route/grpcRouteEmptyMatch (13.41s)
        --- PASS: TestAccAppMesh_serial/Route/http2Route (22.50s)
        --- PASS: TestAccAppMesh_serial/Route/httpHeader (22.34s)
        --- PASS: TestAccAppMesh_serial/Route/disappears (11.52s)
        --- PASS: TestAccAppMesh_serial/Route/http2RouteTimeout (22.48s)
        --- PASS: TestAccAppMesh_serial/Route/tcpRoute (31.74s)
        --- PASS: TestAccAppMesh_serial/Route/grpcRoute (40.62s)
        --- PASS: TestAccAppMesh_serial/Route/grpcRouteWithPortMatch (40.50s)
        --- PASS: TestAccAppMesh_serial/Route/httpRouteWithPortMatch (73.46s)
        --- PASS: TestAccAppMesh_serial/Route/httpRouteTimeout (22.60s)
        --- PASS: TestAccAppMesh_serial/Route/routePriority (22.38s)
        --- PASS: TestAccAppMesh_serial/Route/tags (31.70s)
        --- PASS: TestAccAppMesh_serial/Route/grpcRouteTimeout (22.52s)
        --- PASS: TestAccAppMesh_serial/Route/httpRoute (31.68s)
    --- PASS: TestAccAppMesh_serial/VirtualGateway (253.42s)
        --- PASS: TestAccAppMesh_serial/VirtualGateway/basic (12.36s)
        --- PASS: TestAccAppMesh_serial/VirtualGateway/listenerConnectionPool (20.57s)
        --- PASS: TestAccAppMesh_serial/VirtualGateway/listenerTls (64.23s)
        --- PASS: TestAccAppMesh_serial/VirtualGateway/listenerValidation (20.79s)
        --- PASS: TestAccAppMesh_serial/VirtualGateway/logging (20.86s)
        --- PASS: TestAccAppMesh_serial/VirtualGateway/tags (29.10s)
        --- PASS: TestAccAppMesh_serial/VirtualGateway/disappears (10.58s)
        --- PASS: TestAccAppMesh_serial/VirtualGateway/backendDefaults (20.87s)
        --- PASS: TestAccAppMesh_serial/VirtualGateway/backendDefaultsCertificate (12.50s)
        --- PASS: TestAccAppMesh_serial/VirtualGateway/listenerHealthChecks (20.72s)
        --- PASS: TestAccAppMesh_serial/VirtualGateway/multiListenerValidation (20.84s)
    --- PASS: TestAccAppMesh_serial/VirtualNode (479.17s)
        --- PASS: TestAccAppMesh_serial/VirtualNode/logging (22.07s)
        --- PASS: TestAccAppMesh_serial/VirtualNode/cloudMapServiceDiscovery (108.63s)
        --- PASS: TestAccAppMesh_serial/VirtualNode/listenerHealthChecks (21.15s)
        --- PASS: TestAccAppMesh_serial/VirtualNode/multiListenerValidation (21.01s)
        --- PASS: TestAccAppMesh_serial/VirtualNode/backendClientPolicyFile (21.14s)
        --- PASS: TestAccAppMesh_serial/VirtualNode/tags (29.39s)
        --- PASS: TestAccAppMesh_serial/VirtualNode/listenerOutlierDetection (21.03s)
        --- PASS: TestAccAppMesh_serial/VirtualNode/listenerTimeout (20.84s)
        --- PASS: TestAccAppMesh_serial/VirtualNode/listenerTls (64.28s)
        --- PASS: TestAccAppMesh_serial/VirtualNode/listenerValidation (20.74s)
        --- PASS: TestAccAppMesh_serial/VirtualNode/basic (12.45s)
        --- PASS: TestAccAppMesh_serial/VirtualNode/backendClientPolicyAcm (51.57s)
        --- PASS: TestAccAppMesh_serial/VirtualNode/backendDefaultsCertificate (12.61s)
        --- PASS: TestAccAppMesh_serial/VirtualNode/disappears (10.58s)
        --- PASS: TestAccAppMesh_serial/VirtualNode/backendDefaults (20.76s)
        --- PASS: TestAccAppMesh_serial/VirtualNode/listenerConnectionPool (20.90s)
PASS
ok  	github.com/hashicorp/terraform-provider-aws/internal/service/appmesh	1747.056s
```

**Comment by johnsonaj** [comment]
@dgr237 thank you for the contribution! 👏🏾 🎉

File: internal/service/appmesh/route.go [hunk 1]
```diff
@@ -375,13 +387,35 @@ func ResourceRoute() *schema.Resource {
 																Required:     true,
 																ValidateFunc: validation.IntBetween(0, 100),
 															},
+
+															"port": {
+																Type:         schema.TypeInt,
+																Optional:     true,
+																ValidateFunc: validation.IntBetween(1, 65535),
```
Comment (benmurden) [inline internal/service/appmesh/route.go:394]: This should use already existing `validation.IsPortNumber()`. E.g. https://github.com/hashicorp/terraform-provider-aws/pull/27368/files#diff-344a72a045e6eba371d6884f2e94133929be147a955a19bb342c4018b756660fR498

**Pull request merged**